src/test/scala/unit/inference/SQLInferenceDataStoreSpec.scala

```diff
@@ -586,6 +586,120 @@ trait SQLInferenceRunnerSpec extends FunSpec with BeforeAndAfter { this: SQLInfe
           .map(rs => rs.long("count")).single.apply().get
         assert(numVariablesInc === 20)
       }
+
+      it("should work with materialization and incremental modes (multinomial)") {
+        cancelUnlessPostgres()
+        // XXX Incremental workflow not supported on Postgres-XL as inference.PostgresInferenceRunner.groundVariables raises the following error:
+        //   org.postgresql.util.PSQLException: ERROR: could not plan this distributed update
+        //   Detail: correlated UPDATE or updating distribution column currently not supported in Postgres-XL.
+        if (dataStoreHelper.isUsingPostgresXL) pending
+        // XXX Incremental workflow not supported on Greenplum as datastore.PostgresDataStore.createTableIfNotExists raising following error:
+        //   org.postgresql.util.PSQLException: ERROR: syntax error at or near "NOT"
+        //   Position: 17
+        if (dataStoreHelper.isUsingGreenplum) pending
+
+        inferenceRunner.init()
+
+        // Insert sample data
+        SQL(s"""CREATE TABLE r1(weight text,
+          is_correct int, id bigint);""").execute.apply()
+        val data = (1 to 100).map { i =>
+          Map("id" -> i, "weight" -> s"weight_${i}", "is_correct" -> i%2)
+        }
+        dataStoreHelper.bulkInsert("r1", data.iterator)
+
+        val dbSettingsMat = DbSettings(dbSettings.driver, dbSettings.url, dbSettings.user,
+          dbSettings.password, dbSettings.dbname, dbSettings.host, dbSettings.port,
+          dbSettings.gphost, dbSettings.gppath, dbSettings.gpport, dbSettings.gpload,
+          IncrementalMode.MATERIALIZATION, null)
+
+        val schema = Map[String, VariableDataType]("r1.is_correct" -> MultinomialType(2))
```

Comment:
A multinomial with domain size two may feel like boolean, but it is not the same as boolean. The weights are indexed by variable cardinality. For example, if there's a `weight: ?`, for boolean, it's just one weight, but for multinomial, the number of weights is the variable domain size. 

This test has considered the semantics of multinomial, and the assert conditions are different from boolean case.